lib/logflare/backends/adaptor/postgres/pipeline.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Logflare.Backends.Adaptor.Postgres.Pipeline do
+  @moduledoc false
+  use Broadway
+
+  alias Broadway.Message
+  alias Logflare.Buffers.BufferProducer
+  alias Logflare.Backends.Adaptor.Postgres.LogEvent
+
+  @spec start_link(PostgresAdaptor.t()) :: {:ok, pid()}
+  def start_link(adaptor_state) do
+    Broadway.start_link(__MODULE__,
+      name: adaptor_state.pipeline_name,
+      producer: [
+        module:
+          {BufferProducer,
+           buffer_module: adaptor_state.buffer_module, buffer_pid: adaptor_state.buffer_pid},
+        transformer: {__MODULE__, :transform, []},
+        concurrency: 1
+      ],
+      processors: [
+        default: [concurrency: 3, min_demand: 1]
+      ],
+      context: adaptor_state
+    )
+  end
+
+  # see the implementation for Backends.via_source_backend/2 for how tuples are used to identify child processes
+  def process_name({:via, module, {registry, identifier}}, base_name) do
+    new_identifier = Tuple.append(identifier, base_name)
+    {:via, module, {registry, new_identifier}}
+  end
+
+  def handle_message(_processor_name, message, adaptor_state) do
+    Message.update_data(message, &process_data(&1, adaptor_state))
+  end
+
+  defp process_data(log_event, %{repository_module: repository_module}) do
+    timestamp =
```

Comment:
just wondering out loud - would it be better to store as utc_datetime in db vs naive ?

Reply:
maybe, just currently using the same as `lib/logflare/logs/log_event.ex` to not deviate too much